cameo/flux_analysis/analysis.py

```diff
@@ -367,29 +378,139 @@ class _PhenotypicPhasePlaneChunkEvaluator(object):
     def __init__(self, model, variable_reactions):
         self.model = model
         self.variable_reactions = variable_reactions
+        objective_reactions = [reaction for reaction in model.reactions
+                               if reaction.objective_coefficient != 0]
+        if len(objective_reactions) != 1:
+            raise Exception('multiple objectives not supported')
+        self.product_reaction = objective_reactions[0]
+        self.c_source_reactions = self.get_c_source_reactions()
+
+    def get_c_source_reactions(self):
```

Comment:
Hi guys. This is a tricky and dangerous one.

You are changing the objective of the problem during FVA. When you have two carbon sources, but one cannot be converted into biomass, but it is a precursor of something else, you don't know before hand if something is maximizing just a specific set reactions just by maximizing biomass. Because the total flux is a sum of (n_carbon * flux), you just need to take the *model.medium* reactions *if r.n_carbon > 0*. The overhead is small and avoids problems with evaluating "exotic" designs.

@phantomas1234 @KristianJensen @meono: any thoughts?